### DISCUSSION TITLE
Ignore grammars that failed to fetch when building

### DISCUSSION THREAD

**Discussion by rdbo** [helix-editor/helix#9999] [General]
I was trying to build helix 23.10 to package for my OS, but it wouldn't build because it fails to fetch gemini:
```
  Failure 1/1: gemini Git command failed.
  Stdout: 
  Stderr: fatal: unable to access 'https://git.sr.ht/~sfr/tree-sitter-gemini/': The requested URL returned error: 403


  --- stderr
  thread 'main' panicked at helix-term/build.rs:5:26:
  Failed to fetch tree-sitter grammars: 1 grammars failed to fetch
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

I noticed that there is a way to disable building grammars through the environment var `HELIX_DISABLE_AUTO_GRAMMAR_BUILD`, but that's not what I wanted. I wanted to build helix with grammars, but skip the ones that failed to fetch.
Perhaps a `HELIX_SKIP_UNFETCHED_GRAMMARS` environment var could come handy here.

I also didn't find anything on the current master branch that prevents this from happening in the future, so if a URL goes down again it will fail to build, and the only fixes I can think of would be removing the language or finding a valid URL.

I ended up settling with the master branch instead of 23.10 for convenience (gemini doesn't fail to fetch), but would be nice to avoid this in the future.

**Comment by pascalkuthe** [comment]
* errors should never be silently ignored, it would be confusing if people build helix from source and then some grammars are missing
* The only grammars we ever had issues with are all on sourcehut (which includes gemini). Source hut is just unreliable in general and we now disable all source hut grammars by default as a policy
* The environment variables are a sufficient workaround for cases where you don't want to fetch/build grammars automatically (package manager) and as a workaround

**Reply by rdbo** [reply]
I agree that errors shouldn't be silently ignored, I propose they will be opt-in ignored through an environment variable, for example.
I also agree that the environment variables are good workaround for issues like this. But currently there is no way (that I could find, at least) to continue the build if a grammar fetch fails, which in results makes 23.10 unbuildable (with grammar support, which is the default) without either removing gemini, finding another URL for gemini, or patching the source code. Of course, ideally, there should never be URLs that are unreliable like SourceHut, but you never know when it's gonna happen.
